backend/ibutsu_server/controllers/admin/portal_controller.py

```diff
@@ -0,0 +1,153 @@
+from http import HTTPStatus
+
+import connexion
+from flask import abort
+
+from ibutsu_server.constants import RESPONSE_JSON_REQ
+from ibutsu_server.db.base import session
+from ibutsu_server.db.models import Portal, User
+from ibutsu_server.filters import convert_filter
+from ibutsu_server.util.admin import check_user_is_admin
+from ibutsu_server.util.query import get_offset
+from ibutsu_server.util.uuid import convert_objectid_to_uuid, is_uuid, validate_uuid
+
+
+def admin_add_portal(portal=None, token_info=None, user=None) -> tuple[dict, int]:
+    """Create a portal
+
+    :param body: Portal
+    :type body: dict | bytes
+
+    :rtype: Portal
+    """
+    check_user_is_admin(user)
+    if not connexion.request.is_json:
+        return RESPONSE_JSON_REQ
+    portal = Portal.from_dict(**connexion.request.get_json())
+    # check if portal already exists
+    if portal.id and Portal.query.get(portal.id):
+        return f"Portal id {portal.id} already exist", HTTPStatus.BAD_REQUEST
+    user = User.query.get(user)
+    if user:
+        portal.owner = user
+    session.add(portal)
+    session.commit()
+    return portal.to_dict(), HTTPStatus.CREATED
+
+
+@validate_uuid
+def admin_get_portal(id_, token_info=None, user=None) -> dict:
+    """Get a single portal by ID
+
+    :param id: ID of test portal
+    :type id: str
+
+    :rtype: Portal
+    """
+    check_user_is_admin(user)
+    portal = Portal.query.get(id_)
+    if not portal:
+        portal = Portal.query.filter(Portal.name == id_).first()
+    if not portal:
+        abort(HTTPStatus.NOT_FOUND)
+    return portal.to_dict(with_owner=True)
```

Comment:
To avoid DRY maybe we can use:
```suggestion
    if portal := Portal.query.get(id_) or Portal.query.filter(Portal.name == id_).first():
        return portal.to_dict(with_owner=True)
    abort(HTTPStatus.NOT_FOUND)
```

Reply:
Thanks, I'll take a look at improving there! I copied some patterns that looked unnecessarily verbose and it would be great to condense them here and in the projects controller.

Reply:
Ok. I hate walrus but it's actually useful sometimes

Reply:
I'm learning to like it, and it certainly does make these blocks cleaner.